invenio_search/ext.py

```diff
@@ -93,11 +103,35 @@ def _walk_dir(aliases, *parts):
         # Start the recursion here:
         _walk_dir(self.aliases, alias)
 
-    def load_entry_point_group(self, entry_point_group):
+    def register_templates(self, alias, package_name):
+        """Register templates from a package under given alias.
+
+        :param alias: The alias.
+        :param package_name: The package name.
+        """
+        resource_name = alias
+        for filename in resource_listdir(package_name, resource_name):
+            template_name = build_index_name(filename)
+
+            ext = os.path.splitext(filename)[1]
+            if ext not in {'.json', }:
+                continue
+
+            self.templates[template_name] = resource_filename(
+                package_name, os.path.join(resource_name, filename))
+
+    def load_entry_point_group_mappings(self, entry_point_group_mappings):
         """Load actions from an entry point group."""
-        for ep in iter_entry_points(group=entry_point_group):
+        for ep in iter_entry_points(group=entry_point_group_mappings):
             self.register_mappings(ep.name, ep.module_name)
 
+    def load_entry_point_group_templates(self, entry_point_group_templates):
+        """Load actions from an entry point group."""
+        for ep in iter_entry_points(group=entry_point_group_templates):
+            with self.app.app_context():
+                for template_dir in ep.load()():
```

Comment:
The entry point for the templates returns a callable which in turn returns the directory that contains the templates. It doesn't traverse the directory as it happens for the mappings. This could be changed if needed